cmd/buildkitd/main_containerd_worker.go

```diff
@@ -217,6 +232,13 @@ func containerdWorkerInitializer(c *cli.Context, common workerInitializerOpt) ([
 		return nil, nil
 	}
 
+	if cfg.Rootless {
+		logrus.Debugf("running in rootless mode")
+		if common.config.Workers.Containerd.NetworkConfig.Mode == "auto" {
+			common.config.Workers.Containerd.NetworkConfig.Mode = "host"
```

Comment:
You can use CNI when you have `CAP_NET_ADMIN` in the current namespace, e.g., when RootlessKit is running with slirp4netns/VPNkit/lxc-user-nic/whatever.

https://github.com/rootless-containers/rootlesskit/blob/11642004424410ab18bbbbd195d397f2514c5060/pkg/api/openapi.yaml#L136-L139

Reply:
(OCI worker should be updated to support CNI too, so it can be a separate commit/PR)